File: Leetcode/1638/1638.cpp
class Solution {
public:
    int f(const string &s, const string &t, int i, int j, int n, int m) {
        int left = 0;
        int right = 0;
        int new_i = i;
        int new_j = j;
        // left
        do {
            ++left;
            --new_i;
            --new_j;
        } while (new_i >= 0 && new_j >= 0 && s[new_i] == t[new_j]);
        new_i = i;
        new_j = j;
        // right
        do {
            ++right;
            ++new_i;
            ++new_j;
        } while (new_i < n && new_j < m && s[new_i] == t[new_j]);
        return left*right;
    }
    int countSubstrings(string s, string t) {
        // 中心扩散法
        int sum = 0;
        int n = (int)s.size();
        int m = (int)t.size();
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < m; ++j) {
                if (s[i] != t[j]) {
                    sum += f(s,t,i,j,n,m);
                }
            }
        }
        return sum;
    }
};

File: Leetcode/1638/1638.go
func countSubstrings(s string, t string) int {
	ans := 0
	for i := 0; i < len(s); i++ {
		for j := 0; j < len(t); j++ {
			diff := 0
			for k := 0; i+k < len(s) && j+k < len(t); k++ {
				if s[i+k] != t[j+k] {
					diff++
				}
				if diff > 1 {
					break
				} else if diff == 1 {
					ans++
				}
			}
		}
	}
	return ans
}